echopype/echodata/combine.py

```diff
@@ -79,385 +142,274 @@ def check_and_correct_reversed_time(
     ValueError
-        If EchoData objects have conflicting source file names.
+        If the provided ``EchoData`` objects have the same filenames
+    RuntimeError
+        If the first time value of each ``EchoData`` group is not less
+        than the first time value of the subsequent corresponding
+        ``EchoData`` group, with respect to the order in ``echodatas``
+    RuntimeError
+        If the same ``EchoData`` groups in ``echodatas`` do not
+        have the same number of channels and the same name for each
+        of these channels.
+    RuntimeError
+        If any of the following attribute checks are not met
+        amongst the combined ``EchoData`` groups:
+
+        - the keys are not the same
+        - the values are not identical
+        - the keys ``date_created`` or ``conversion_time``
+          do not have the same types
 
     Warns
     -----
     UserWarning
-        If the ``sonar_model`` of the input ``EchoData`` objects is ``"EK60"`` and any
-        ``EchoData`` objects have non-monotonically increasing ``ping_time``, ``time1``
-        or ``time2`` values, the corresponding values in the output ``EchoData`` object
-        will be increased starting at the timestamp where the reversal occurs such that
-        all values in the output are monotonically increasing. Additionally, the original
-        ``ping_time``, ``time1`` or ``time2`` values will be stored in the ``Provenance``
-        group, although this behavior may change in future versions.
-
-    Warnings
-    --------
-    Changes in parameters between ``EchoData`` objects are not currently checked;
-    however, they may raise an error in future versions.
+        If any time coordinate in a final combined group is not
+        in ascending order (see Notes below for more details).
 
     Notes
     -----
-    * ``EchoData`` objects are combined by combining their groups individually.
-    * Attributes from all groups before the combination will be stored in the provenance group,
-      although this behavior may change in future versions.
-    * The ``source_file`` and ``converted_raw_path`` attributes will be copied from the first
-      ``EchoData`` object in the given list, but this may change in future versions.
+    * ``EchoData`` objects are combined by appending their groups individually to a zarr store.
+    * All attributes (besides array attributes) from all groups before the combination will be
+      stored in the ``Provenance`` group.
+    * The instance attributes ``source_file`` and ``converted_raw_path`` of the combined
+      ``EchoData`` object will be copied from the first ``EchoData`` object in the given list.
+    * If any time coordinate in a final combined group is not in ascending order, then it will
+      be corrected according to `#297 <https://github.com/OSOceanAcoustics/echopype/pull/297>`_.
+      # TODO: change this to description of the gist of the employed correction.
```

Comment:
@leewujung I accepted and committed this change, but I didn't realize that it was in the middle of a docstring. I think we should place this somewhere else because it will show up in the generated docs for users.

Reply:
Yes, I agree, perhaps move it to immediately below the entire """ block so we would remember (I'll linked this comment in #798). Looking at this again I am thinking if we have a section in the docs that explain how we do the correction, we can put that link into this docstring here.

Reply:
> Looking at this again I am thinking if we have a section in the docs that explain how we do the correction, we can put that link into this docstring here.

Yes, that is the best solution.